driver-mapping/src/main/java/com/datastax/driver/mapping/DefaultPropertyMapper.java

```diff
@@ -0,0 +1,542 @@
+        return this;
+    }
+
+    /**
+     * Sets the {@link PropertyTransienceStrategy property transience strategy} to use.
+     * The default is {@link PropertyTransienceStrategy#OPT_OUT}.
+     *
+     * @param propertyTransienceStrategy the {@link PropertyTransienceStrategy property transience strategy} to use; may not be {@code null}.
+     * @return this {@link DefaultPropertyMapper} instance (to allow for fluent builder pattern).
+     */
+    public DefaultPropertyMapper setPropertyTransienceStrategy(PropertyTransienceStrategy propertyTransienceStrategy) {
+        this.propertyTransienceStrategy = checkNotNull(propertyTransienceStrategy);
+        return this;
+    }
+
+    /**
+     * Sets the {@link HierarchyScanStrategy hierarchy scan strategy} to use.
+     * The default is {@link DefaultHierarchyScanStrategy}.
+     *
+     * @param hierarchyScanStrategy the {@link HierarchyScanStrategy hierarchy scan strategy} to use; may not be {@code null}.
+     * @return this {@link DefaultPropertyMapper} instance (to allow for fluent builder pattern).
+     */
+    public DefaultPropertyMapper setHierarchyScanStrategy(HierarchyScanStrategy hierarchyScanStrategy) {
+        this.hierarchyScanStrategy = checkNotNull(hierarchyScanStrategy);
+        return this;
+    }
+
+    /**
+     * Sets the {@link NamingStrategy naming strategy} to use.
+     * The default is {@link DefaultNamingStrategy}.
+     *
+     * @param namingStrategy the {@link NamingStrategy naming strategy} to use; may not be {@code null}.
+     * @return this {@link DefaultPropertyMapper} instance (to allow for fluent builder pattern).
+     */
+    public DefaultPropertyMapper setNamingStrategy(NamingStrategy namingStrategy) {
+        this.namingStrategy = checkNotNull(namingStrategy);
+        return this;
+    }
+
+    /**
+     * Sets transient property names. This
+     * will completely replace any names already configured for this object.
+     * <p/>
+     * The default set comprises the following property names:
+     * {@code class} and {@code metaClass}.
+     * These properties pertain to the {@link Object} class –
+     * {@code metaClass} being specific to the Groovy language.
+     * <p/>
+     * Property names provided here will always be considered transient;
+     * if a more fine-grained tuning is required, it is also possible
+     * to use the {@link Transient @Transient} annotation
+     * on a specific property.
+     * <p/>
+     * Subclasses can also override {@link #isTransient(String, Field, Method, Method, Map)} to gain
+     * complete control over which properties should be considered transient.
+     *
+     * @param transientPropertyNames a set of property names to exclude from mapping; may not be {@code null}. This
+     *                               will completely replace any names already configured for this object.
+     */
+    public DefaultPropertyMapper setTransientPropertyNames(Set<String> transientPropertyNames) {
```

Comment:
That feels less commonly needed, and as a last resort you can override `isTransient`.

I think what was called `WHITE_LIST` at some point was actually the equivalent of `PropertyTransienceStrategy.OPT_IN`.